src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -90,6 +94,26 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
         Picture picture = personToEdit.getPicture().orElse(null);
         List<SpecialDate> dates = personToEdit.getDates();
         List<Meeting> meetings = personToEdit.getMeetings();
+
+        Optional<Meeting> earliestMeeting = meetings.stream().min(Comparator.comparing(Meeting::getDate));
+        Optional<SpecialDate> earliestDate = dates.stream().min(Comparator.comparing(SpecialDate::getDate));
```

Comment:
Aren't meetings and dates already sorted on add? The comparison here might be unnecessary.

Reply:
Oh good catch

Reply:
Actually, Lin Lue's one was okay. If the json data that is read not sorted, the bug will occur again but harder to find now because it is dependent on the order of the dates/meetings in the json.

Reply:
Actually nvm, the `specialDates` and `meetings` are already sorted in the `JsonAdaptedPerson`